tests/test_keep_alive_timeout.py

```diff
@@ -0,0 +1,269 @@
+from json import JSONDecodeError
+from sanic import Sanic
+import asyncio
+from asyncio import sleep as aio_sleep
+from sanic.response import text
+from sanic.config import Config
+from sanic import server
+import aiohttp
+from aiohttp import TCPConnector
+from sanic.testing import SanicTestClient, HOST, PORT
+
+
+class ReuseableTCPConnector(TCPConnector):
+    def __init__(self, *args, **kwargs):
+        super(ReuseableTCPConnector, self).__init__(*args, **kwargs)
+        self.old_proto = None
+
+    @asyncio.coroutine
+    def connect(self, req):
+        new_conn = yield from super(ReuseableTCPConnector, self)\
+                                                         .connect(req)
+        if self.old_proto is not None:
+            if self.old_proto != new_conn._protocol:
+                raise RuntimeError(
+                    "We got a new connection, wanted the same one!")
+        print(new_conn.__dict__)
+        self.old_proto = new_conn._protocol
+        return new_conn
+
+
+class ReuseableSanicTestClient(SanicTestClient):
```

Comment:
Is there any reason we need this ? can we use `test_client` fixture in `pytest-sanic` ? https://github.com/yunstanford/pytest-sanic, it'd be better to maintain.

Reply:
This is here because I needed a custom version of the `SanicTestClient` that could:
1) Honor a keepalive header, and store the connection in the client for reuse.
2) Reuse the existing TCP connection when making a subsequent request, if keepalive was set on the first request.
3) Raise an error if we got a new connection, when we were expecting to reuse the old connection.

These customisations to the `SanicTestClient` are very specific to this particular test (the KeepAliveTimeout test)

I haven't looked, but I don't think the `test_client` fixture in `pytest-sanic` has the ability to be customised in this manner.

Reply:
No, you can do them all, by simply passing your `ReuseableTCPConnector ` as param to `test_client`.